task-list/app/controllers/tasks_controller.rb

```diff
@@ -12,20 +12,43 @@ def show
   end
 
   def new
-    @task = Task.new
+      @task = Task.new
+      @action = "create"
+      @title = "New Task"
   end
 
   def create
     Task.create(task_params[:task])
-    redirect_to "/"
+    redirect_to '/tasks/'
   end
 
   def destroy
     id = params[:id]
     Task.destroy(id)
-    redirect_to "/"
+    redirect_to '/tasks/'
   end
-  
+
+  def set_completed
+    Task.update(params[:id], :completed_at => Time.now)
+    redirect_to '/tasks/'
+  end
+
+  def update
+    task = Task.find(params[:id])
+    task.update(name: task_params[:task][:name], description: task_params[:task][:description])
```

Comment:
Because you have written the `task_params` method which uses `params.permit` you can actually simplify this line a bit. Try this instead: `task.update(task_params[:task])`.